prov/util/src/util_av.c

```diff
@@ -948,21 +950,53 @@ static int ip_av_lookup(struct fid_av *av_fid, fi_addr_t fi_addr, void *addr,
 	return 0;
 }
 
-static const char *ip_av_straddr(struct fid_av *av, const void *addr,
-				  char *buf, size_t *len)
+static const char *ip_av_straddr(struct fid_av *av, const void *addr, char *buf,
+				 size_t *len)
 {
+	const struct sockaddr *sock_addr;
+	const struct sockaddr_in6 *sin6;
+	const struct sockaddr_in *sin;
 	char str[INET6_ADDRSTRLEN + 8];
-	size_t size;
+	uint16_t port;
+	int size;
+
+	if (!addr || !len)
+		return NULL;
+
+	sock_addr = addr;
+	switch (sock_addr->sa_family) {
+	case AF_INET:
+		sin = addr;
+		if (!inet_ntop(sin->sin_family, &sin->sin_addr.s_addr, str,
+			       sizeof(str)))
+			return NULL;
+
+		port = sin->sin_port;
+		break;
+	case AF_INET6:
+		sin6 = addr;
+		if (!inet_ntop(sin6->sin6_family, &sin6->sin6_addr.s6_addr, str,
+			       sizeof(str)))
+			return NULL;
 
-	if (!inet_ntop(((struct sockaddr *) addr)->sa_family, addr,
-			str, sizeof str))
+		port = sin6->sin6_port;
+		break;
+	default:
 		return NULL;
+	}
 
-	size = strlen(str);
-	size += snprintf(&str[size], sizeof(str) - size, ":%d",
-			 ((struct sockaddr_in *) addr)->sin_port);
-	memcpy(buf, str, MIN(*len, size));
+	size = snprintf(buf, MIN(*len, sizeof(str)), "%s:%" PRIu16, str,
```

Comment:
ok - changes look fine then.  I've just never used anything other than PRI_64, but it sounds like the others are standard.